wallet_migrator/src/lib.rs

```diff
@@ -0,0 +1,67 @@
+pub mod credx2anoncreds;
+pub mod error;
+pub mod vdrtools2credx;
+
+use std::fmt::Display;
+
+use error::MigrationResult;
+use log::{debug, error, info, warn};
+pub use vdrtools::{
+    types::domain::wallet::{Config, Credentials, KeyDerivationMethod, Record},
+    Locator, WalletHandle,
+};
+
+/// Retrieves all records from a wallet and migrates them
+/// by applying the `migrate_fn` argument.
+///
+/// The migrated records are inserted into a newly created
+/// wallet, based on the provided `config` and `credentials`.
+pub async fn migrate_wallet<E>(
+    wallet_handle: WalletHandle,
+    config: Config,
+    credentials: Credentials,
+    migrate_fn: impl FnMut(Record) -> Result<Option<Record>, E>,
+) -> MigrationResult<()>
+where
+    E: Display,
+{
+    info!("Starting wallet migration...");
+    let locator = Locator::instance();
+
+    debug!("Creating new wallet {} ...", config.id);
+    locator
+        .wallet_controller
+        .create(config.clone(), credentials.clone())
+        .await?;
+
+    debug!("Opening new wallet...");
+
+    let new_wh = locator
+        .wallet_controller
+        .open(config.clone(), credentials.clone())
+        .await?;
+
+    debug!("Migrating records from wallet with handle {wallet_handle:?} to wallet with handle {new_wh:?}");
+
+    let res = locator
+        .wallet_controller
+        .migrate_records(wallet_handle, new_wh, migrate_fn)
+        .await;
+
+    if let Err(e) = locator.wallet_controller.close(new_wh).await {
+        error!("Closing the wallet failed: {e}");
+    }
+
+    if let Err(e) = &res {
+        error!("Migration error encountered: {e}");
+
+        match locator.wallet_controller.delete(config, credentials).await {
```

Comment:
if you agree on https://github.com/hyperledger/aries-vcx/pull/867/files#r1248167895 then I'd suggest to omit any deletion of wallets as well - since we leave creation of wallets up to them, it would be consistent to let them handle deletion (if they want to) as well

Reply:
I don't really see any benefit in it, really. But it might be just me.